module-apps/apps-common/locks/handlers/PhoneLockHandler.cpp

```diff
@@ -127,12 +130,12 @@ namespace locks
         owner->bus.sendMulticast(std::make_shared<locks::UnlockedPhone>(), sys::BusChannel::PhoneLockChanges);
     }
 
-    void PhoneLockHandler::phoneInputRequiredAction()
+    void PhoneLockHandler::phoneInputRequiredAction(bool isSoftRequest = false)
```

Comment:
Don't use bool as a function parameter. 
I recommend splitting the function or using an enum instead. 
I will look better and will be easier to understand

Reply:
Ok, you right.
I changed bool to enum `PhoneLockHandler::TypeOfRequest`

Maybe code is still not clear enough and I should split functions?

Reply:
I would consider renaming it "PhoneLockHandler::TypeOfRequest" 
As I noticed it has 2 values Soft and Normal and for example, TypeOfRequest::Soft is not saying much about what it is and what it does
Try something that will explain this -> why unlock for MTP is soft? (UnlockType:: {UnlockPhone, UnlockMTPOnly} 
`
connect(typeid(locks::UnlockPhone), [&](sys::Message *request) -> sys::MessagePointer {
            return phoneLockHandler.handleUnlockRequest(locks::PhoneLockHandler::TypeOfRequest::Normal);
        });
        connect(typeid(locks::UnlockPhoneForMTP), [&](sys::Message *request) -> sys::MessagePointer {
            return phoneLockHandler.handleUnlockRequest(locks::PhoneLockHandler::TypeOfRequest::Soft);
        });
`
if it doesn't help much try to consider the introduction of a second function dedicated only to unlocking MTP

Reply:
Ok then
I changed _enum_ names to:
```cpp
        enum class ReasonForRequest
        {
            Unspecified = 0, // For default behaviors
            UnlockForMTP
        };
 ```
 and added a proper comment where passing of the _new enum_  as arg. actually changes the behavior:
 ```cpp
        // If it's for MTP, don't show the input passcode popup if it's already on the screen
        data->ignoreIfTheSamePopupIsOnTopOfTheStack = (reqReason == ReasonForRequest::UnlockForMTP);
```

Reply:
> Unspecified = 0, // For default behaviors

So maybe default? :D